experimental/displaynames/tests/tests.rs

```diff
@@ -0,0 +1,197 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[allow(unused_imports)]
+use icu_displaynames::{DisplayNamesOptions, Fallback, LanguageDisplay, Style, Capitalization, LocaleDisplayNamesFormatter};
+use writeable::assert_writeable_eq;
+use icu_locid::locale;
+
+macro_rules! generate_locale_displaynames_test {
+    ($test_name: ident, $constructor: ident, $options: expr,
+     [$(($locale:literal, $locale_to_test:literal, $expected:literal)),+ $(,)?]) => {
+        #[test]
+        fn $test_name(){
+            let display_name_obj = LocaleDisplayNamesFormatter::$constructor(
+                &icu_testdata::unstable(),
+                &locale!($locale).into(),
+                $options
+            )
+            .expect("Data should load successfully.");
+
+            $(
+                assert_writeable_eq!(
+                    display_name_obj.of(locale!($locale_to_test)).unwrap_or_default(),
+                    $expected
+                );
+            )+
+        }
+    };
+}
+generate_locale_displaynames_test!(
```

Comment:
Suggestion (optional): avoid using macros in test cases by following the following pattern:

https://github.com/unicode-org/icu4x/blob/main/utils/fixed_decimal/src/decimal.rs#L3343